views/ListCurrencies.tsx

```diff
@@ -0,0 +1,90 @@
+import React, {useState, useEffect, FC} from 'react';
+import { useDispatch } from 'react-redux';
+import { useSelector } from 'react-redux';
+import { StyleSheet, Text, TouchableOpacity, View, Image, ScrollView } from 'react-native';
+
+import CurrencyCard from '../components/CurrencyCard';
+import { Currency } from '../interfaces';
+import currenciesActions from '../redux/actions/currenciesActions';
+import { MainState } from '../redux/reducers/mainReducer';
+
+const Home: FC = ({navigation}: any) => {
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    dispatch(currenciesActions.getUserCurrenciesFromAsync())
+  }, [])
+  
+  const userCurrenciesList = useSelector((state: MainState) => state.currencies.userCurrenciesList)
+  
+  return (
+    <ScrollView contentContainerStyle={styles.mainContainer}>
+      <View style={styles.headerContainer}>
+        <Text style={styles.headerTitle}>CryptoTracker Pro</Text>
+        <Image style={styles.userPhoto} source={{uri: 'https://i.imgur.com/kregEJT.png'}} />
+      </View>
+      <View style={{paddingHorizontal: 24}}>
+        {(userCurrenciesList && userCurrenciesList.length === 0) && <View style={styles.noCryptoContainer}>
+            <Text style={styles.noCryptoText}>You haven't added cryptocurrencies yet</Text>
+          </View>
+        }
+        {userCurrenciesList && userCurrenciesList.map((currency: Currency) => <CurrencyCard key={currency.symbol} currency={currency} />)}
```

Comment:
[CRITICAL] Never use .map() to render a list with unknown length; this can have CRITICAL performance issues on any scalable app. Please switch to using a FlatList instead of ScrollView.